include/seqan3/alphabet/concept_detail.hpp

```diff
@@ -0,0 +1,220 @@
+constexpr underlying_rank_t<alphabet_type> to_rank(alphabet_type const a)
+    requires requires (alphabet_type a) { { a.to_rank() } -> underlying_rank_t<alphabet_type>; }
+{
+    return a.to_rank();
+}
+
+/*!\brief Implementation of
+ * \link alphabet_concept && seqan3::assign_char(alphabet_concept && a, char_type const c)\endlink
+ * that delegates to a member function.
+ * \tparam alphabet_type Must provide an `.assign_char()` member function.
+ * \param a The alphabet letter that you wish to assign to.
+ * \param c The `char` value you wish to assign.
+ * \returns A reference to the alphabet letter you passed in.
+ */
+template <typename alphabet_type>
+constexpr alphabet_type & assign_char(alphabet_type & a, underlying_char_t<alphabet_type> const c)
+    requires requires (alphabet_type a) { { a.assign_char(char{0}) } -> alphabet_type &; }
+{
+    return a.assign_char(c);
+}
+
+/*!\brief Implementation of
+ * \link alphabet_concept && seqan3::assign_char(alphabet_concept && a, char_type const c)\endlink
+ * that delegates to a member function.
+ * \tparam alphabet_type Must provide an `.assign_char()` member function.
+ * \param a An alphabet letter temporary.
+ * \param c The `char` value you wish to assign.
+ * \returns The assignment result as a temporary.
+ * \details
+ * Use this e.g. to newly create alphabet letters from char:
+ * ~~~{.cpp}
+ * auto l = assign_char(dna5{}, 'G');  // l is of type dna5
+ * ~~~
+ */
+template <typename alphabet_type>
+constexpr alphabet_type && assign_char(alphabet_type && a, underlying_char_t<alphabet_type> const c)
+    requires requires (alphabet_type a) { { a.assign_char(char{0}) } -> alphabet_type &; }
+{
+    return std::move(a.assign_char(c));
+}
+
+/*!\brief Implementation of
+ * \link semi_alphabet_concept && seqan3::assign_rank(semi_alphabet_concept && a, rank_type const i) \endlink
+ * that delegates to a member function.
+ * \tparam alphabet_type Must provide an `.assign_rank()` member function.
+ * \param a The alphabet letter that you wish to assign to.
+ * \param i The `rank` value you wish to assign.
+ * \returns A reference to the alphabet letter you passed in.
+ */
+template <typename alphabet_type>
+constexpr alphabet_type & assign_rank(alphabet_type & a, underlying_rank_t<alphabet_type> const i)
+    requires requires (alphabet_type a) { { a.assign_rank(uint8_t{0}) } -> alphabet_type &; }
+{
+    return a.assign_rank(i);
+}
+
+/*!\brief Implementation of
+ * \link semi_alphabet_concept && seqan3::assign_rank(semi_alphabet_concept && a, rank_type const i) \endlink
+ * that delegates to a member function.
+ * \tparam alphabet_type Must provide an `.assign_rank()` member function.
+ * \param a An alphabet letter temporary.
+ * \param i The `rank` value you wish to assign.
+ * \returns The assignment result as a temporary.
+ * \details
+ * Use this e.g. to newly create alphabet letters from rank:
+ * ~~~{.cpp}
+ * auto l = assign_rank(dna5{}, 1);  // l is of type dna5 and == dna5::C
```

Comment:
hm, I am not convinced, but in any case, it would be part of polishing the nucleotide submodule and not part of this PR.